pkg/provisioner/controller.go

```diff
@@ -638,3 +655,43 @@ func (c *obcController) objectBucketForClaimKey(key string) (*v1alpha1.ObjectBuc
 	}
 	return ob, nil
 }
+
+func (c *obcController) handleUpdateClaim(key string, obc *v1alpha1.ObjectBucketClaim, class *storagev1.StorageClass) error {
+	ob, err := getObjectBucketFromClaimKey(key, obc, c.libClientset, defaultRetryBaseInterval, defaultRetryTimeout)
+	if err != nil {
+		log.Error(err, "getting ob failed", "key", key)
+		return err
+	}
+	if reflect.DeepEqual(ob.Spec.Endpoint.AdditionalConfigData, obc.Spec.AdditionalConfig) {
+		return nil
+	}
+	log.Info("Additional config changed, hence updating OB")
+	tmp := make(map[string]string)
+
+	overwriteAdditionalConfig(ob.Spec.Endpoint.AdditionalConfigData, tmp)
```

Comment:
Suggest delete L669 and here replace with:
`tmp := ob.Spec.Endpoint.AdditionalConfigData`